openai/openai_test.go

```diff
@@ -82,6 +82,38 @@ func TestMiddleware(t *testing.T) {
 				}
 			},
 		},
+		{
+			Name:     "completions handler",
```

Comment:
We could probably use 2-3 more cases:

1. invalid types
2. params: temperature etc

Reply:
invalid types is a little weird to simulate with middleware since the struct won't allow me to pass in an invalid type. for the user invalid types are caught by (un)marshaling